packages/store/src/helpers/json-path-utils.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * Parses a JSON Pointer string into an array of path segments
+ *
+ * @example
+ * ```ts
+ * parseJsonPointer('#/components/schemas/User')
+ *
+ * ['components', 'schemas', 'User']
+ * ```
+ */
+export function parseJsonPointer(pointer: string): string[] {
+  return (
+    pointer
+      // Split on '/'
+      .split('/')
+      // Remove the leading '#' if present
+      .filter((segment, index) => index !== 0 || segment !== '#')
+  )
+}
```

Comment:
do we need to unescape the pointer here? I think I’d expect this funciton to return the unescaped strings here